scvi/inference/posterior.py

```diff
@@ -150,7 +148,17 @@ def save_posterior(self, dir_path: str):
         anndata_dataset = self.gene_dataset.to_anndata()
 
         anndata_dataset.write(os.path.join(dir_path, "anndata_dataset.h5ad"))
+        with open(os.path.join(dir_path, "posterior_type.txt"), "w") as post_file:
+            post_file.write(self.posterior_type)
         torch.save(self.model.state_dict(), os.path.join(dir_path, "model_params.pt"))
+
+        # Saves posterior indices and kwargs that can easily be retrieved
+        data_loader_kwargs = pd.Series(
+            {key: vals for key, vals in self.data_loader_kwargs.items()}
```

Comment:
does this store the sampler? "sampler" is one of the keys in data_loader_kwargs no? I thought we were trying to avoid storing the sampler.

Reply:
we should also avoid storing the "collate_fn" key.